### PR TITLE
Regenerate swagger, bump version, and default EndpointSuffix to 'core.windows.net'

### LINKED ISSUE #16510: Update the autorest.md file to use swagger README instead of json file
related to #16445

### LINKED ISSUE #18554: Regenerate table client since swagger was updated
https://github.com/Azure/azure-rest-api-specs/blob/3227f9795b09642afb975f40d45ed481192b43a2/specification/cosmos-db/data-plane/Microsoft.Tables/preview/2019-02-02/table.json

### LINKED ISSUE #18557: Bump Tables client to version 12

### LINKED ISSUE #18949: [BUG] Connection String parsing fails in Azure.Data.Tables TableClient and TableServiceClient
**Describe the bug**
The Azure.Data.Tables library (TableConnectionString) does not correctly parse connections strings if they do not have a TableEndpoint and do not have a EndpointSuffix

**Expected behavior**
The same connection string parses just fine when used for accessing blobs (BlobClient etc.). The blob libraries just use the default EndpointSuffix (core.windows.net)

**Actual behavior (include Exception or Stack Trace)**
Exception is thrown in TableConnectionString

**To Reproduce**
Pass a connection string to TableClient or TableServiceLCinet etc. that does not have a TableEndpoint nor an EndpointSuffix.

**Environment:**
 - Azure.Data.Tables 3.0.0-beta.5
 - .net Core 5

### PR DESCRIPTION
**Pull request by christothes**
fixes #18949
closes #18557
closes #18554
closes #16510

### CODE REVIEW

File: sdk/core/Azure.Core/src/Shared/ConnectionString.cs [hunk 1]
```diff
@@ -37,8 +37,16 @@ private ConnectionString(Dictionary<string, string> pairs, string pairSeparator,
         public string? GetNonRequired(string keyword) =>
             _pairs.TryGetValue(keyword, out var value) ? value : null;
 
-        public bool TryGetSegmentValue(string keyword, out string? value) =>
-            _pairs.TryGetValue(keyword, out value);
+        public bool TryGetSegmentValue(string keyword, out string? value, string? defaultValue = null)
```
Comment (jsquire) [inline sdk/core/Azure.Core/src/Shared/ConnectionString.cs:40]: Is there a test in `Azure.Core` that we can extend to cover the new functionality - or is there an existing fixture that we can add to?

**Review by jsquire — APPROVED** [review]

File: sdk/core/Azure.Core/src/Shared/ConnectionString.cs (hunk 1 above)
Comment (christothes) [reply sdk/core/Azure.Core/src/Shared/ConnectionString.cs:40]: Good callout - I'll add some coverage.

Comment (pakrym) [reply sdk/core/Azure.Core/src/Shared/ConnectionString.cs:40]: Let's add another method instead. 

`GetSegmentValueOrDefault()`

When using the default value the return value is always `true` and the method is just more awkward to use.

``` C#
var defauledValue = CS.GetSegmentValueOrDefault("Segment", "Default");
//vs
CS.TryGetSegmentValue("Segment", out defauledValue , "Default");
```

Comment (christothes) [reply sdk/core/Azure.Core/src/Shared/ConnectionString.cs:40]: That's fair - I was being lazy :)

**Review by pakrym — APPROVED** [review]

**Pull request merged**